pandas/core/index.py

```diff
@@ -1219,14 +1219,17 @@ def union(self, other):
         if len(other) == 0 or self.equals(other):
             return self
 
+        other = _ensure_index(other)
+
```

Comment:
don't change something unrelated (and this is in a fast path for other things, so it causes a perf drag)